### DISCUSSION TITLE
access input from parallelforeach using csharp script

### DISCUSSION THREAD

**Discussion by mremadevi** [elsa-workflows/elsa-core#5326] [Q&A]
How to access input from parallelforeach using csharp script

**Comment by glime-ai** [comment]
To access input from a `ParallelForEach` activity using C# script in Elsa Workflow, you would typically use the `Input` property provided by the `ActivityExecutionContext`. The `Input` property allows you to retrieve the input that was passed to the workflow or to a specific activity.

Here's an example of how you might access an input named "MyInput" within a C# script in an activity that is part of a `ParallelForEach`:

```csharp
var myInputValue = context.Input.Get("MyInput");
```

In this example, `context` is an instance of `ActivityExecutionContext`, which is provided to the activity's `ExecuteAsync` method. The `Get` method is used to retrieve the value of the input with the specified key.

If you are using JavaScript expressions instead of C# scripts, you would use the `getInput` function like this:

```javascript
var myInputValue = getInput("MyInput");
```

Please note that the exact method to access the input may vary depending on the version of Elsa Workflow you are using and the context in which you are trying to access the input. If you encounter any issues or have specific requirements, please provide more details so I can give you a more precise answer.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
